src/Sylius/Behat/Context/Ui/Admin/ManagingChannelsContext.php

```diff
@@ -0,0 +1,482 @@
+use Sylius\Component\Core\Model\ChannelInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Łukasz Chruściel <lukasz.chrusciel@lakion.com>
+ */
+final class ManagingChannelsContext implements Context
+{
+    const RESOURCE_NAME = 'channel';
+
+    /**
+     * @var IndexPageInterface
+     */
+    private $indexPage;
+
+    /**
+     * @var CreatePageInterface
+     */
+    private $createPage;
+
+    /**
+     * @var UpdatePageInterface
+     */
+    private $updatePage;
+
+    /**
+     * @var CurrentPageResolverInterface
+     */
+    private $currentPageResolver;
+
+    /**
+     * @var NotificationCheckerInterface
+     */
+    private $notificationChecker;
+
+    /**
+     * @param IndexPageInterface $indexPage
+     * @param CreatePageInterface $createPage
+     * @param UpdatePageInterface $updatePage
+     * @param CurrentPageResolverInterface $currentPageResolver
+     * @param NotificationCheckerInterface $notificationChecker
+     */
+    public function __construct(
+        IndexPageInterface $indexPage,
+        CreatePageInterface $createPage,
+        UpdatePageInterface $updatePage,
+        CurrentPageResolverInterface $currentPageResolver,
+        NotificationCheckerInterface $notificationChecker
+    ) {
+        $this->indexPage = $indexPage;
+        $this->createPage = $createPage;
+        $this->updatePage = $updatePage;
+        $this->currentPageResolver = $currentPageResolver;
+        $this->notificationChecker = $notificationChecker;
+    }
+
+    /**
+     * @Given I want to create a new channel
+     */
+    public function iWantToCreateANewChannel()
+    {
+        $this->createPage->open();
+    }
+
+    /**
+     * @When I specify its code as :code
+     * @When I do not specify its code
+     */
+    public function iSpecifyItsCodeAs($code = null)
+    {
+        $this->createPage->specifyCode($code);
+    }
+
+    /**
+     * @When I name it :name
+     * @When I rename it to :name
+     * @When I do not name it
+     * @When I remove its name
+     */
+    public function iNameIt($name = null)
+    {
+        $this->createPage->nameIt($name);
+    }
+
+    /**
+     * @When I add it
+     * @When I try to add it
+     */
+    public function iAddIt()
+    {
+        $this->createPage->create();
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully created
+     */
+    public function iShouldBeNotifiedItHasBeenSuccessfulCreation()
+    {
+        $this->notificationChecker->checkCreationNotification(self::RESOURCE_NAME);
+    }
+
+    /**
+     * @Then the channel :channelName should appear in the registry
+     * @Then the channel :channelName should be in the registry
+     * @Then /^(this channel) should still be in the registry$/
+     */
+    public function theChannelShouldAppearInTheRegistry($channelName)
+    {
+        $this->iWantToBrowseChannels();
+
+        Assert::true(
+            $this->indexPage->isResourceOnPage(['name' => $channelName]),
```

Comment:
And we assume, that channel has `_toString()` that returns its name, but it's not a part of channel interface